File: src/main/java/com/github/pagviewer/nativebridge/PagPreviewInfo.java
package com.github.pagviewer.nativebridge;

public record PagPreviewInfo(
        int width,
        int height,
        int frameCount,
        float frameRate,
        int compositionWidth,
        int compositionHeight
) {
    public boolean isRenderable() {
        return width > 0 && height > 0 && frameCount > 0 && frameRate > 0.0f;
    }

    public boolean hasScaledDecoderSize() {
        return compositionWidth > 0
                && compositionHeight > 0
                && (compositionWidth != width || compositionHeight != height);
    }
}

File: src/main/kotlin/com/github/pagviewer/nativebridge/PagPreviewInfo.kt
package com.github.pagviewer.nativebridge

data class PagPreviewInfo(
    val width: Int,
    val height: Int,
    val frameCount: Int,
    val frameRate: Float,
    val compositionWidth: Int,
    val compositionHeight: Int
) {
    fun isRenderable(): Boolean =
        width > 0 && height > 0 && frameCount > 0 && frameRate > 0.0f

    fun hasScaledDecoderSize(): Boolean =
        compositionWidth > 0 &&
            compositionHeight > 0 &&
            (compositionWidth != width || compositionHeight != height)
}
